### DISCUSSION TITLE
Can I set the required chain according to the wallet that is connecting to my dapp?

### DISCUSSION THREAD

**Discussion by einaralex** [rainbow-me/rainbowkit#1574] [Q&A] [answered]
For example, in our test environment, we support Goerli, Mumbai and Chiado.
However, some wallets that want to use our service through WalletConnect only support Mumbai.
Can I detect the chain of wallet requests and only require that when creating the session?

**Comment by DanielSinclair** [comment, marked as answer]
With recent version of the WalletConnect v2 library, most wallets (if they're on the ball) should support something called `optional namespaces`. This means that you can pair your dApp with a wallet that doesn't support all of your chains. Proper support in wallets should solve your core issue there. But you can also always using Wagmi's `useNetwork` and `useSwitchNetwork` hooks to better gate behavior in your dApp.
